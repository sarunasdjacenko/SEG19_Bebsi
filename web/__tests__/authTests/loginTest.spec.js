import { shallowMount } from "@vue/test-utils";
import Vue from "vue";
import VueRouter from "vue-router";
// @ts-ignore
import Component from "../../src/components/auth/Login.vue";
Vue.use(VueRouter);

describe("Component", () => {
  const wrapper = shallowMount(Component);
  test("is a Vue instance", () => {
    expect(wrapper.isVueInstance()).toBeTruthy();
  }),
    //snapshot test
    test("renders correctly", () => {
      expect(wrapper.element).toMatchSnapshot();
    }),
    test("has the correct buttons rendered", () => {
      const loginBtn = wrapper.find("#loginBtn");
      const resetPasswordBtn = wrapper.find("#resetPasswordBtn");

      expect(resetPasswordBtn).toBeDefined();
      expect(loginBtn).toBeDefined();
    }),
    test("form submits successfully", () => {
      wrapper.setMethods({ login: jest.fn() });
      wrapper.find("#loginBtn").trigger("click");

      expect(wrapper.vm.login).toHaveBeenCalled();
    }),
    test("has the right amount of inputs", () => {
        expect(wrapper.findAll("input")).toHaveLength(2);
    });
});
