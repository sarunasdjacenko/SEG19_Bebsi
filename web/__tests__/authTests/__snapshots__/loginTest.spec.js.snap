// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component renders correctly 1`] = `
<div
  id="login"
>
  <div
    class="container"
    style="padding-top: 10em;"
  >
    <div
      class="row"
    >
      <div
        class="col s12 m8 offset-m2"
      >
        <div
          class="login card-panel white blue-text center"
        >
          <h3>
            Login
          </h3>
           
          <form>
            <div
              class="input-field"
            >
              <i
                class="material-icons prefix"
              >
                email
              </i>
               
              <input
                id="email"
                name="email"
                placeholder="Email"
                type="text"
              />
            </div>
             
            <div
              class="input-field"
            >
              <i
                class="material-icons prefix"
              >
                lock
              </i>
               
              <input
                id="password"
                name="password"
                placeholder="Password"
                type="password"
              />
            </div>
             
            <button
              class="btn btn-large blue lighten-2 black-text"
              id="loginBtn"
            >
              Login
            </button>
             
            <router-link-stub
              event="click"
              id="resetPasswordBtn"
              style="margin-left: 50px;"
              tag="a"
              to="/resetPassword"
            >
              Forgot Password?
            </router-link-stub>
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;
